### ISSUE TITLE
Can't be build in F-Droid due to binaries

### ISSUE THREAD

**Issue by jspricke** [lucasnlm/antimine-android#261]
Your app fails to build on F-Droid:
```
2021-04-06 09:05:25,015 ERROR: Found shared library at gdx/libs/armeabi/libgdx.so
2021-04-06 09:05:25,015 ERROR: Found shared library at gdx/libs/x86_64/libgdx.so
2021-04-06 09:05:25,016 ERROR: Found shared library at gdx/libs/armeabi-v7a/libgdx.so
2021-04-06 09:05:25,016 ERROR: Found shared library at gdx/libs/x86/libgdx.so
2021-04-06 09:05:25,016 ERROR: Found shared library at gdx/libs/arm64-v8a/libgdx.so
```
https://monitor.f-droid.org/builds/log/dev.lucanlm.antimine/1000001

Is there a way to build it without those?

**Comment by lucasnlm** [comment]
They are required. However they are commited in `gdx` module [here](https://github.com/lucasnlm/antimine-android/tree/master/gdx/libs).

Is it possible that some F-droid script delete them before the build?

**Comment by jspricke** [comment]
> They are required.
> Is it possible that some F-droid script delete them before the build?

I'm not sure I understand you. For F-Droid everything needs to be build from source, we don't accept binaries build by someone else. That's why the build complains about binaries and fails. I understand that those binaries are required, so would there be a way to build them from source? Or are they not required and we could delete them before building?

**Comment by lucasnlm** [comment]
I mean they are required to build and I can't get them from source.
Currently I get them from `"com.badlogicgames.gdx:gdx-backend-android:1.9.14"` [here](https://github.com/lucasnlm/antimine-android/blob/0d1d85c520c403c69aedbf7567774b82c02b8592/gdx/build.gradle#L58)

And then, calling `copyAndroidNatives` gradle method to copy the native libraries.

Is it possible to add this `copyAndroidNatives` step on F-droid build script?

**Comment by jspricke** [comment]
If we can't build them from source we can't include it in the F-Droid repo. I will disable the app then.

**Comment by lucasnlm** [comment]
Can we let it until version 9.1.1?

**Comment by jspricke** [comment]
yes

**Comment by lucasnlm** [comment]
One last question. Do you know if it's possible to let F-droid get LibGdx libraries from cache somehow?
It doesn't make sense to me that `com.google.android.material:material:1.3.0` works with f-droid but libgdx doesn't.

**Comment by jspricke** [comment]
`com.google.android.material:material:1.3.0` is imported from mavenCentral, which is trusted by F-Droid: https://f-droid.org/de/docs/Build_Metadata_Reference/#build_srclibs
What do you mean by 'cache'? As written above, we would need to build LibGdx from source.

**Comment by lucasnlm** [comment]
@jspricke I just published a new release (10.0.3) without that `.so` libraries. Can we try again with that version?

**Comment by jspricke** [comment]
That change would work for me, I've opened a MR: https://gitlab.com/fdroid/fdroiddata/-/merge_requests/8761
Thanks!
